### DISCUSSION TITLE
how to load a spectrum from a dataset with more than 2 columns with spectrum.from_local_file?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#72] [Q&A] [answered]
Normally spectra is saved as two columns (Raman shift, wavelength, etc for the x axis and Intensity for the y axis). However, some software from some systems, for example in the case of the Neegala system of Elodiz, the files are saved with the value for different parameters, followed by a number of columns (I give an example further down). Is there already an option implemented in ramanchada2 to load spectra from such files? I can easily circunvent the problem now that I am looking to a single spectra, but it would be convenient to address this in the future.

Example:
id,4175c5d9-b564-45f3-b27f-28a19be22e38
name,PST_Neegala785_LongProbe_100%_60sx4_Position0_001
device,HAH504
spectrometer,ZTFZ
time_stamp,2023-02-10 10:21:35
integ_time,60
integ_unit,s
averages,4
laserPower,100
user,Admin
wl,785nm
Pixels,Wavelength,Wavenumbers,Raman_Shift,Raw_Data,Background_Data,Processed_Data
0,735.492993,9453.13756960459,-858.2247997601046,1208,1194,14
1,735.74773028291,9453.692410099477,-853.4119950192265,1208,1194,14
2,736.0023706674255,9454.248132884524,-848.6048070612062,1141,1166,-25
...

**Comment by georgievgeorgi** [comment, marked as answer]
RC2 v0.0.3 should be able to read these files.
```python
import ramanchada2 as rc2
rc2.spectrum.from_local_file('<your-neegala-file.txt>')
```
It takes `Wavenumbers` as x-axis and `Processed_Data` as y-axis. It might be beneficial to use also the rest of the data form the file, but this was never discussed so not implemented.
If there is a problem, it should be fixed.
Let me know

**Reply by nicocopez** [reply]
Thank you for the quick reply Georgi.

I get the following error: 
ValueError: Some errors were detected !
    Line #5 (got 2 columns instead of 1)

Besides, is there any reason why do you choose Wavenumbers? I would asume people working in Raman spectroscopy would go for Raman_shift, or Wavelengths if working with luminescence.

**Reply by georgievgeorgi** [reply]
can you make sure that `rc2.__version__` is `'0.0.3'`

**Reply by nicocopez** [reply]
It took my time, but finally was able to update the version and `rc2.spectrum.from_local_file('<your-neegala-file.txt>')` did the job, using the Raman shift, for what I can see. Thank you Georgi! :)
